src/AssimpWorker/import/ColladaImporter/ColladaRecursiveImporter.cpp

```diff
@@ -0,0 +1,156 @@
+/*
+* This file is part of ATLAS. It is subject to the license terms in
+* the LICENSE file found in the top-level directory of this distribution.
+* (Also avialable at http://www.apache.org/licenses/LICENSE-2.0.txt)
+* You may not use this file except in compliance with the License.
+*/
+
+#include <iostream>
+#include "ColladaRecursiveImporter.hpp"
+#include "../../internal/Exception.hpp"
+
+namespace AssimpWorker {
+
+	using ATLAS::Model::DataDeletingBlob;
+	using ATLAS::Model::Folder;
+
+	ColladaRecursiveImporter::ColladaRecursiveImporter(const Poco::URI& colladaFileURI, Log& log, const std::string& pathToWorkingDirectory, ColladaMassagerRegistry& registry, float scale) :
```

Comment:
This constructor seems to have too many arguments. I feel like a version with `const ColladaRecursiveImporter& parent` could work as well.